### PR TITLE
Migrate to CircleCI 2.0

### PR DESCRIPTION
**Pull request by abesto**
Part of https://github.com/openzipkin/zipkin/issues/2102

## Diff on CircleCI 1.0 config against `zipkin-aws` pre-migration:

```diff
# diff zipkin-aws/circle.yml brave-karaf/circle.yml
15a16,19
>   java:
>     version: openjdk8
>   services:
>     - mysql
17c21
<     JAVA_HOME: /usr/lib/jvm/zulu-9-amd64
---
>     MYSQL_USER: root
21,24d24
<     - sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 0x219BD9C9
<     - sudo apt-add-repository -y "deb http://repos.azulsystems.com/ubuntu stable main"
<     - sudo apt-get -y update
<     - sudo apt-get -y install zulu-9
30c30
<     - ./mvnw verify
---
>     - ./mvnw verify install
56a57
>     # triples the timeout to 30 minutes as maven central sync takes a lot longer than 10m
58c59,60
<       - ./build-support/publish-stable.sh
---
>       - ./build-support/publish-stable.sh:
>         timeout: 1800
```

 * we're on Java 8
 * an extra `mvn install`
 * longer timeout for the publish step
 * MySQL (though we only run unit tests, no integration tests, so that's a bit confusing)

## Diff on CircleCI 2.0 config in this PR against `zipkin-aws`:

```diff
# diff {zipkin-aws,brave-karaf}/.circleci/config.yml
19c19,26
<       - image: circleci/openjdk:9-jdk
---
>       - image: circleci/openjdk:8-jdk
>         environment:
>           MYSQL_USER: root
> 
>       - image: circleci/mysql:5.7
>         environment:
>           MYSQL_USER: root
> 
38c45
<           command: ./mvnw clean verify
---
>           command: ./mvnw clean verify install
51c58
<       - image: circleci/openjdk:9-jdk
---
>       - image: circleci/openjdk:8-jdk
66c73
<       - image: circleci/openjdk:9-jdk
---
>       - image: circleci/openjdk:8-jdk
72a80,81
>           # triples the timeout to 30 minutes as maven central sync takes a lot longer than 10m
>           no_output_timeout: 30m
81c90
<       - image: circleci/openjdk:9-jdk
---
>       - image: circleci/openjdk:8-jdk
```

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+            sudo apt-get install xsltproc
+            ./build-support/go-offline.sh
+      - save_cache:
+          key: maven-dependencies-{{ checksum "pom.xml" }}
+          paths:
+            - ~/.m2
+
+      - run:
+          name: Tests
+          command: ./mvnw clean verify install
```
Comment (shakuzen) [inline .circleci/config.yml]: Nit: the Maven phase `install` will execute before the `verify` phase so we don't need to write both; just `verify` should be equivalent

**Review by shakuzen — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
also I agree. One thing is that this repo doesnt' use mysql and also it was incorrectly copy/pasted from zipkin-aws (#5 changes the name where copy/paste was incorrect)

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+version: 2
+jobs:
+  build:
+    docker:
+      - image: circleci/openjdk:8-jdk
+        environment:
+          MYSQL_USER: root
+
+      - image: circleci/mysql:5.7
```
Comment (codefromthecrypt) [inline .circleci/config.yml]: we can remove this as it doesn't use mysql

**Comment by abesto** [comment]
I've addressed removing MySQL and getting rid of `mvn install`. I'm unclear on what's going on with #5, and about what was incorrectly copy-pasted; does that mean we should first merge #5? Or rebase this on top of that?

Also: test failed now, which is weird. Didn't expect these changes to actually _change_ anything.

**Comment by cschneider** [comment]
The test fails as the karaf feature is not available in the maven repo. Can you try adding mvn install again.

**Comment by codefromthecrypt** [comment]
getting rid of mvn install maybe needs to be undone?

**Comment by shakuzen** [comment]
> The test fails as the karaf feature is not available in the maven repo. Can you try adding mvn install again.


> getting rid of mvn install maybe needs to be undone?

Oops. I had a brain fart. `install` phase is AFTER `verify`, not before. Sorry for the confusion. So the command could be updated to `./mvnw clean install` instead of `./mvnw clean verify`

Ref: https://maven.apache.org/guides/introduction/introduction-to-the-lifecycle.html

**Comment by cschneider** [comment]
:-)

**Comment by codefromthecrypt** [comment]
thanks for the help, all. The other PR will be in conflict. At some point I will re-adjust it if no-one beats me to it

**Comment by codefromthecrypt** [comment]
that point won't be today :P

**Comment by abesto** [comment]
Thanks for finishing this up!

**Pull request merged**